### PR TITLE
HIVE-21588: Remove HBase dependency from hive-metastore

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

HIVE-17234 has removed HBase metastore from master. But maven dependency have not been removed. We should remove it.

### Why are the changes needed?

Reduce dependency to make Spark dependency more clear:
https://github.com/apache/spark/blob/3ce4ab545bfc28db7df2c559726b887b0c8c33b7/pom.xml#L1997-L2007

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

File: ql/pom.xml [hunk 1]
```diff
@@ -820,6 +820,12 @@
       <version>${guava.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>it.unimi.dsi</groupId>
+      <artifactId>fastutil</artifactId>
+      <version>6.5.6</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (wangyum) [inline ql/pom.xml:828]: https://github.com/apache/hive/blob/4b670877c280b37c5776046f66d766079489b2a8/ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java#L20 need this dependency:
image

**Comment by wangyum** [comment]
Retest this please.

**Comment by wangyum** [comment]
cc @alanfgates  @sunchao

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java [hunk 2]
```diff
@@ -17,7 +17,6 @@
  */
 package org.apache.hadoop.hive.ql.txn.compactor;
 
-import it.unimi.dsi.fastutil.booleans.AbstractBooleanBidirectionalIterator;
```
Comment (sunchao) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java:20]: is this related?

**Review by sunchao — APPROVED** [review]
LGTM except one nit.

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java (hunk 2 above)
Comment (wangyum) [reply ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java:20]: Yes, this need `it.unimi.dsi:fastutil`, we have removed this dependency:
image

Comment (sunchao) [reply ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java:20]: Gotcha, cool.

**Comment by sunchao** [comment]
Merged. Thanks @wangyum !
